### PR TITLE
Yafi/profile

### PR DESCRIPTION
**Pull request by YafiAhsanH**
### Background: 
- create profile page
- create edit profile page
- create User cubit
- integrate profile page with backend
- enable deep linking

### Additional Packages/Dependencies:
-

### Changes:
- android/app/src/main/AndroidManifest.xml
- ios/Runner/Info.plist
- lib/app.dart
- lib/auth/data/auth_repository.dart
- lib/auth/user/bloc/user_cubit.dart
- lib/auth/user/bloc/user_state.dart
- lib/auth/user/data/user_repository.dart
- lib/common/components/forms/custom_date_picker.dart
- lib/common/components/forms/custom_form_input_class.dart
- lib/common/components/forms/custom_text_field.dart
- lib/common/components/forms/form_component.dart
- lib/page/edit_profile.dart
- lib/page/profile.dart
- lib/page/splash.dart

### How to Implement:
-

### CODE REVIEW

File: lib/page/edit_profile.dart [hunk 1]
```diff
@@ -72,23 +74,65 @@ class _EditProfileState extends State<EditProfile> {
...
+            fontSize: 20.0,
             fontWeight: FontWeight.bold,
-            color: Theme.of(context).colorScheme.tertiary,
+            color: Theme.of(context).colorScheme.onSurfaceVariant,
           ),
         ),
-        const SizedBox(
-          height: 7.0,
-        ),
+        // TextButton(
```
Comment (RafadanaM) [inline lib/page/edit_profile.dart]: remove comment?

File: lib/page/profile.dart [hunk 2]
```diff
@@ -12,54 +18,69 @@ class Profile extends StatefulWidget {
...
+        body: Container(
+          decoration:
+              BoxDecoration(color: Theme.of(context).colorScheme.secondary),
+          child: SafeArea(
+            child: BlocBuilder<UserCubit, UserState>(
+              builder: (context, state) {
+                if (state is UserLoadingState) {
+                  return const Center(
+                    child: CircularProgressIndicator(
+                      color: Colors.amber,
```
Comment (RafadanaM) [inline lib/page/profile.dart]: change the color

File: lib/page/profile.dart [hunk 3]
```diff
@@ -12,54 +18,69 @@ class Profile extends StatefulWidget {
...
+                    ),
+                  );
+                } else if (state is UserErrorState) {
+                  return Text(state.errorMessage);
+                } else if (state is UserSuccessState) {
+                  return buildProfile(state.user);
+                } else {
+                  return const Text("initial");
```
Comment (RafadanaM) [inline lib/page/profile.dart]: maybe leave it blank?

File: lib/page/profile.dart [hunk 4]
```diff
@@ -221,9 +245,9 @@ class _ProfileState extends State<Profile> {
     );
   }
 
-  List<Widget> buildInterest() {
+  List<Widget> buildInterest(List<PreferenceModel> preferences) {
     List<Widget> widgets = [];
-    for (String interest in interests) {
+    for (PreferenceModel preference in preferences) {
```
Comment (RafadanaM) [inline lib/page/profile.dart]: use map() instead of adding them to widgets

**Review by RafadanaM — CHANGES_REQUESTED** [review]
Please check the comments for changes

File: lib/page/profile.dart (hunk 4 above)
Comment (YafiAhsanH) [reply lib/page/profile.dart]: done, now it uses map

**Review by oliver-mf — APPROVED** [review]
looks gud!

**Review by RafadanaM — APPROVED** [review]
Looks good to me!

**Pull request merged**